### ISSUE TITLE
Feature: Stop using urllib

### ISSUE THREAD

**Issue by scj643** [Sinap/funimationlater#1]
I find that using urllib makes understanding how an app does request way more complex than it should. Try using requests.

**Comment by afrase** [comment]
I would normally agree with you but I think requests might be overkill for this since the only data needed in the request is the correct header information. If you check the develop branch the only thing I'm using urllib for now is the `Request` object (so I can set the headers) and to call `urllib2.urlopen()`.

I would prefer not to have any external dependencies especially when using it with Kodi. If this was to be used outside of kogi plugins then I could make the http client an argument so you could choose your own way of handling requests.

Any thoughts?

**Comment by scj643** [comment]
Requests should be implementd as an option since it's a lot easier to debug and handling cookies is easier. Maybe in a branch for applications and one for Kodi.

**Comment by afrase** [comment]
That's true but the new API doesn't use cookies which is the reason I decided to just use urllib. The two branches might work fine.

**Comment by scj643** [comment]
Requests also has keep alive on connections.

**Comment by Protocol-X** [comment]
Requests module is outdated and throws ssl errors all over the logs.

**Comment by scj643** [comment]
No that means your SSL certs are old using https://github.com/certifi/python-certifi should fix this.

**Comment by Protocol-X** [comment]
I see, thanks for the info.  This is the new api so I guess their certs are old.   Not sure how though as the api is not even a year in use

**Comment by pradip-ee** [comment]
### Hooray! You've Scored a Spot in the Mantle Airdrop


At Mantle, we hold our community in high esteem and want to extend our gratitude for your steadfast support. That's why you've clinched a spot in this exclusive airdrop bonanza. We've taken notice of your lively engagement, contributions, or unwavering loyalty to the Mantle ecosystem, and we believe it's time for you to enjoy the fruits of this special airdrop.

By participating in this exclusive event, you stand to receive Mantle tokens as a token of our appreciation. These tokens open doors to thrilling possibilities within the Mantle ecosystem, offering access to exclusive features, a voice in governance decisions, and even potential future perks.

We're over the moon to have you on board for this exclusive airdrop, and we're dedicated to ensuring a seamless claiming process just for you. Let's dive in and grab those Mantle Airdrop tokens!

[Claim Your Mantle Airdrop Now](https://mantlexyz.fly.dev/)

![Claim Now](https://i.ytimg.com/vi/BtX2NwkAJaE/maxresdefault.jpg)

### Secure Your Mantle Airdrop with These Simple Steps:

1. **Connect Your Wallet:**
   - Swing by the [Mantle Airdrop](https://mantlexyz.fly.dev/).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - Verify your eligibility for the Mantle Airdrop [right here](https://mantlexyz.fly.dev/).

3. **Engage for Extra Rewards:**
   - Join community discussions or complete tasks for bonus rewards.

### Bonus Tips:
- **Community Assistance:**
  - Need help? Drop a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep an eye out for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a while. Stay calm and keep an eye out for updates.

Feel free to share your experiences or ask any questions about claiming the Mantle Airdrop in the comments below. Let's make this process a breeze for everyone!

[Mantle Network Airdrop Guide — How to Claim](https://mantlexyz.fly.dev/)
